examples/custom-ipfs-repo/index.js

```diff
@@ -0,0 +1,99 @@
+'use strict'
+
+const IPFS = require('ipfs')
+const Repo = require('ipfs-repo')
+const fsLock = require('ipfs-repo/src/lock')
+
+// Create our custom options
+const customRepositoryOptions = {
+
+  /**
+   * IPFS nodes store different information in separate storageBackends, or datastores.
+   * Each storage backend can use the same type of datastore or a different one — you
+   * could store your keys in a levelDB database while everything else is in files,
+   * for example. (See https://github.com/ipfs/interface-datastore for more about datastores.)
+   */
+  storageBackends: {
+    root: require('datastore-fs'), // version and config data will be saved here
+    blocks: require('datastore-fs'),
+    keys: require('datastore-fs'),
+    datastore: require('datastore-fs')
+  },
+
+  /**
+   * Storage Backend Options will get passed into the instantiation of their counterpart
+   * in `storageBackends`. If you create a custom datastore, this is where you can pass in
+   * custom constructor arguments. You can see an S3 datastore example at:
+   * https://github.com/ipfs/js-datastore-s3/tree/master/examples/full-s3-repo
+   */
+  storageBackendOptions: {
+    root: {
+      extension: '.ipfsroot', // Used by datastore-fs; Appended to all files
+      errorIfExists: false, // Used by datastore-fs; If the datastore exists, don't throw an error
+      createIfMissing: true // Used by datastore-fs; If the datastore doesn't exist yet, create it
+    },
+    blocks: {
+      sharding: false, // Used by IPFSRepo Blockstore to determine sharding; Ignored by datastore-fs
+      extension: '.ipfsblock',
+      errorIfExists: false,
+      createIfMissing: true
+    },
+    keys: {
+      extension: '.ipfskey',
+      errorIfExists: false,
+      createIfMissing: true
+    },
+    datastore: {
+      extension: '.ipfsds',
```

Comment:
No extension used by default

Reply:
The reason I added the non-default options was to demonstrate that configuration options to the users. Perhaps I could add in some comments about the defaults here and mention that they are being overridden for demonstration purposes only?

Reply:
Yeah, this would probably be useful to call that out here.

It might also be good to, in the last step of the script, do something like:

```js
.then(() => {
  console.log('Check "/tmp/custom-repo/.ipfs" to see what your customized repository looks like on disk.')
}
```